tree/tree/src/InternalTreeUtils.cxx

```diff
@@ -23,6 +23,53 @@ namespace ROOT {
 namespace Internal {
 namespace TreeUtils {
 
+////////////////////////////////////////////////////////////////////////////////
+/// \brief Add information of a single friend.
+///
+/// \param[in] treeName Name of the tree.
+/// \param[in] fileNameGlob Path to the file. Refer to TChain::Add for globbing rules.
+/// \param[in] alias Alias for this friend.
+void RFriendInfo::AddFriend(const std::string &treeName, const std::string &fileNameGlob, const std::string &alias)
+{
+   fFriendNames.emplace_back(std::make_pair(treeName, alias));
+   fFriendFileNames.emplace_back(std::vector<std::string>{fileNameGlob});
+   fFriendChainSubNames.emplace_back();
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// \brief Add information of a single friend.
+///
+/// \param[in] treeName Name of the tree.
+/// \param[in] fileNameGlobs Paths to the files. Refer to TChain::Add for globbing rules.
+/// \param[in] alias Alias for this friend.
+void RFriendInfo::AddFriend(const std::string &treeName, const std::vector<std::string> &fileNameGlobs,
+                            const std::string &alias)
+{
+   fFriendNames.emplace_back(std::make_pair(treeName, alias));
+   fFriendFileNames.emplace_back(fileNameGlobs);
+   fFriendChainSubNames.emplace_back(std::vector<std::string>(fileNameGlobs.size(), treeName));
```

Comment:
can't `fFriendChainSubNames` take only a single `fileNameGlobs`? I think that only affect internals, hence it would be safe to change the current behaviour, no?

Reply:
It makes it more difficult to write
```cpp
TChain thisfriend;
for (std::size_t i = 0; i < fFriendFileNames.size(); i++){
    std::string fullpath = fFriendFileNames[i] + "?#" + fFriendChainSubNames[i];
}
```
Unfortunately we don't have `zip_longest` in C++

Reply:
Yes, that is an option, I dislike all the extra `if`s. I would already like to remove the `if` that is currently there when we check if the friend is a TTree/TChain. But I recognise this is just my preference and not something strictly needed

Reply:
alright, I would agree to keep it as it is now. We can optimize in future